deepchem/feat/molecule_featurizers/mat_featurizer.py

```diff
@@ -0,0 +1,96 @@
+from deepchem.feat.base_classes import MolecularFeaturizer
+try:
+  import logging
+  import numpy as np
+  import pandas as pd
+  import torch
+  from rdkit import Chem
+  from rdkit.Chem import AllChem
+  from rdkit.Chem import MolFromSmiles
+  from sklearn.metrics import pairwise_distances
+  from torch.utils import datasets
+except:
+  raise ImportError("Required Modules not found.")
+
+use_cuda = torch.cuda.is_available()
+
+if use_cuda:
+  FloatTensor = torch.cuda.FloatTensor
+  LongTensor = torch.cuda.LongTensor
+  IntTensor = torch.cuda.IntTensor
+  DoubleTensor = torch.cuda.DoubleTensor
+else:
+  FloatTensor = torch.FloatTensor
+  LongTensor = torch.LongTensor
+  IntTensor = torch.cuda.IntTensor
+  DoubleTensor = torch.cuda.DoubleTensor
+
+
+class MATFeaturizer(MolecularFeaturizer):
+
+  def __init__(
+      self,
+      add_dummy_node=True,
+      one_hot_formal_charge=True,
+  ):
+    self.mol = mol
+    self.add_dummy_node = add_dummy_node
+    self.one_hot_formal_charge = one_hot_formal_charge
+
+  def OHVector(value, array):
+    if value not in array:
+      value = array[-1]
+    return map(lambda x: x == value, array)
+
+  def atom_features(self, atom, one_hot_formal_charge=True):
+    attrib = []
+    attrib += OHVector(atom.getAtomicNum(),
+                       [5, 6, 7, 8, 9, 15, 16, 17, 35, 53, 999])
+    attrib += OHVector(len(atom.GetNeighbors()), [0, 1, 2, 3, 4, 5])
+    attrib += OHVector(atom.GetTotalNumHs(), [0, 1, 2, 3, 4])
+
+    if one_hot_formal_charge:
+      attrib += OHVector(atom.GetFormalCharge(), [-1, 0, 1])
+    else:
+      attrib.append(atom.GetFormalCharge())
+
+    attrib.append(atom.IsInRing())
+    attrib.append(atom.GetIsAromatic())
+
+    return np.array(attrib, dtype=np.float32)
+
+  def _featurize(self, mol, add_dummy_node, one_hot_formal_charge):
+    node_features = np.array(
+        [atom_features(atom, one_hot_formal_charge) for atom in mol.getAtoms()])
+    adjacency_matrix = np.eye(mol.GetNumAtoms())
```

Comment:
`adjacency_matrix = Chem.rdmolops.GetAdjacencyMatrix(mol)`. A function in rdkit already exists for adjacency matrix creation.

Reply:
Done! I have also replaced the distance matrix calculation with the inbuilt rdkit.Chem.molops.GetDistanceMatrix(mol) function. I did not know about these earlier, thank you!